server/datastore/mysql/mysql.go

```diff
@@ -484,6 +489,11 @@ func (ds *Datastore) HealthCheck() error {
 
 // Close frees resources associated with underlying mysql connection
 func (ds *Datastore) Close() error {
+	if ds.loadHostStmt != nil {
```

Comment:
Hmm this is a race with the Once that writes to it... One option that is ugly but simple would be to call the Once before this (it doesn't even have to do anything, just call it with a no-op func), just so it doesn't get written again concurrently.

Reply:
This would not be an issue if we move the prepared statement to the constructor, see https://github.com/fleetdm/fleet/pull/4128#discussion_r803090493.